src/repository/user.repository.ts

```diff
@@ -0,0 +1,105 @@
+import User from '@domain/entities/user.js';
+import UserStorage from './storage/user.storage.js';
+import GoogleApiTransport from '@repository/transport/google-api/index.js';
+import { GetUserInfoResponsePayload } from '@repository/transport/google-api/types/GetUserInfoResponsePayload';
+
+/**
+ * OAuth provider
+ */
+export enum Provider {
+  /**
+   * Google provider
+   */
+  GOOGLE = 'google',
+}
+
+/**
+ * Repository allows accessing data from business-logic (domain) level
+ */
+export default class UserRepository {
+  /**
+   * User storage instance
+   */
+  public storage: UserStorage;
+
+  /**
+   * Google API transport
+   */
+  private readonly googleApiTransport: GoogleApiTransport;
+
+  /**
+   * User repository constructor
+   *
+   * @param storage - storage for user
+   * @param googleApiTransport - google api transport
+   */
+  constructor(storage: UserStorage, googleApiTransport: GoogleApiTransport) {
+    this.googleApiTransport = googleApiTransport;
+    this.storage = storage;
+  }
+
+  /**
+   * Get user by provider
+   *
+   * @param accessToken - provider access token
+   * @param provider - provider
+   * @returns { Promise<User | null> } found user
+   */
+  public async getUserByProvider(accessToken: string, provider: Provider): Promise<User | null> {
+    let res: GetUserInfoResponsePayload;
+
+    /**
+     * Set request headers
+     */
+    const headers = {
```

Comment:
headers should be in the transport layer

Reply:
done